platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py

```diff
@@ -365,9 +365,47 @@ def get_high_critical_threshold(self):
             up to nearest thousandth of one degree Celsius, e.g. 30.125
         """
         if self.high_critical_threshold is None:
-            return self.high_crit_th_default
+            return None
+
+        if not self._retry_until_file_exists(self.high_critical_threshold):
+            return None
+
         value = utils.read_float_from_file(self.high_critical_threshold, None, log_func=logger.log_info)
-        return value / self.scale if (value is not None and value != 0) else self.high_crit_th_default
+        return value / 1000.0 if (value is not None and value != 0) else None
+
+    def _retry_until_file_exists(self, file_path):
```

Comment:
blocking call? not sure this is what we want

Reply:
it waits up to 5 seconds for each thermal object. During the wait, thermalctld will be blocked.

Reply:
Agree, removed wait here.